src/alarm_trap_sender.cpp

```diff
@@ -43,167 +43,24 @@
-
-bool AlarmFilter::alarm_filtered(unsigned int index, unsigned int severity)
-{
-  unsigned long now = current_time_ms();
-
-  if (now > _clean_time)
-  {
-    _clean_time = now + CLEAN_FILTER_TIME;
-
-    std::map<AlarmFilterKey, unsigned long>::iterator it = _issue_times.begin();
-    while (it != _issue_times.end())
-    {
-      if (now > (it->second + ALARM_FILTER_TIME))
-      {
-        _issue_times.erase(it++);
-      }
-      else
-      {
-        ++it;
-      }
-    }
-  }
-
-  AlarmFilterKey key(index, severity);
-  std::map<AlarmFilterKey, unsigned long>::iterator it = _issue_times.find(key);
-
-  bool filtered = false;
-
-  if (it != _issue_times.end())
-  {
-    if (now > (it->second + ALARM_FILTER_TIME))
-    {
-      _issue_times[key] = now;
-    }
-    else
-    {
-      filtered = true;
-    }
-  }
-  else
-  {
-    _issue_times[key] = now;
-  }
-
-  return filtered;
-}
-
-bool AlarmFilter::AlarmFilterKey::operator<(const AlarmFilter::AlarmFilterKey& rhs) const
-{
-  return  (_index  < rhs._index) ||
-         ((_index == rhs._index) && (_severity < rhs._severity));
-}
-
-unsigned long AlarmFilter::current_time_ms()
-{
-  struct timespec ts;
-
-  clock_gettime(CLOCK_MONOTONIC, &ts);
-
-  return ts.tv_sec * 1000 + (ts.tv_nsec / 1000000);
-}
-
-void AlarmTrapSender::issue_alarm(const std::string& issuer, const std::string& identifier)
-{
-  unsigned int index;
-  unsigned int severity;
-  if (sscanf(identifier.c_str(), "%u.%u", &index, &severity) != 2)
-  {
-    TRC_ERROR("malformed alarm identifier: %s", identifier.c_str());
-    return;
-  }
-
-  AlarmTableDef& alarm_table_def = AlarmTableDefs::get_instance().get_definition(index, severity);
-
-  if (alarm_table_def.is_valid())
-  {
-    // If the alarm to be issued exists in the ObservedAlarms mapping at the
-    // same severity then we disregard the alarm as it has not changed state.
-    if (_observed_alarms.update(alarm_table_def, issuer))
-    {
-      if (!AlarmFilter::get_instance().alarm_filtered(index, severity))
-      {
-        alarmActiveTable_trap_handler(alarm_table_def);
-        send_trap(alarm_table_def);
-      }
-    }
-  }
-  else
-  {
-    TRC_ERROR("unknown alarm definition: %s", identifier.c_str());
-  }
-}
-
-void AlarmTrapSender::sync_alarms()
-{
-  for (ObservedAlarmsIterator it = _observed_alarms.begin(); it != _observed_alarms.end(); it++)
-  {
-    send_trap(it->alarm_table_def());
-  }
-}
-
 static int alarm_trap_send_callback(int op,
                                     snmp_session* session,
                                     int req_id,
                                     snmp_pdu* pdu,
                                     void* correlator)
 {
-  AlarmTableDef* alarm_table_def = (AlarmTableDef *)correlator; correlator = NULL;
-  AlarmTrapSender::get_instance().alarm_trap_send_callback(op,
-                                                           *alarm_table_def);
+  ((SNMPCallbackAlarmInfo*)correlator)->_ats->alarm_trap_send_callback(
```

Comment:
Clearer as two lines?

```
SNMPCallbackAlarmInfo* cbi = static_cast<SNMPCallbackAlarmInfo*>(correlator);
cbi->_ats->alarm_trap_send_callback(op, cbi->_atd);
```

Also, _ats and _atd are a bit obscure - `_sender` and `_definition`?